indra/sources/hprd/processor.py

```diff
@@ -0,0 +1,350 @@
+            # is obsolete; take the UP ID from HGNC
+            if len(up_ids) == 0:
+                self.no_up_for_refseq.append(refseq_id)
+                up_id = up_id_from_hgnc
+            # More than one reviewed entry--no thanks, we'll take the one from
+            # HGNC instead
+            elif len(up_ids) > 1:
+                self.many_ups_for_refseq.append(refseq_id)
+                up_id = up_id_from_hgnc
+            # We got a unique, reviewed UP entry for the RefSeq ID
+            else:
+                up_id = up_ids[0]
+                # If it's the canonical isoform, strip off the '-1'
+                if up_id.endswith('-1'):
+                    up_id = up_id.split('-')[0]
+        # For completeness, get the Refseq ID from the HPRD ID table
+        else:
+            refseq_id = self.id_df.loc[hprd_id].REFSEQ_PROTEIN
+            up_id = up_id_from_hgnc
+        # Make db_refs, return Agent
+        db_refs = {'HGNC': hgnc_id, 'UP': up_id, 'EGID': egid,
+                   'REFSEQ_PROT': refseq_id}
+        return Agent(hgnc_name, db_refs=db_refs)
+
+    def _get_evidence(self, hprd_id, isoform_id, pmid_str, evidence_type,
+                      info_type, motif_dict=None):
+        pmids = pmid_str.split(',')
+        ev_list = []
+        if motif_dict is None:
+            motif_dict = {}
+        for pmid in pmids:
+            ev_type_list = [] if evidence_type is nan \
+                               else evidence_type.split(';')
+            # Add the annotations with the site motif info if relevant
+            annotations = {'evidence': ev_type_list}
+            annotations.update(motif_dict)
+            ev = Evidence(source_api='hprd',
+                          source_id=_hprd_url(hprd_id, isoform_id, info_type),
+                          pmid=pmid, annotations=annotations)
+            ev_list.append(ev)
+        return ev_list
+
+    def _get_seq_motif(self, refseq_id, residue, pos_str):
+        seq = self.seq_dict[refseq_id]
+        pos_1ix = int(pos_str)
+        pos_0ix = pos_1ix - 1
+        if seq[pos_0ix] != residue:
+            self.invalid_site_pos.append((refseq_id, residue, pos_str))
+            if seq[pos_0ix + 1] == residue:
+                self.off_by_one.append((refseq_id, residue, pos_str))
+                motif, respos = \
+                          ProtMapper.motif_from_position_seq(seq, pos_1ix + 1)
+                return {'site_motif': {'motif': motif, 'respos': respos,
+                                       'off_by_one': True}}
+            else:
+                return {}
+        else:
+            # The index of the residue at the start of the window
+            motif, respos =\
+                        ProtMapper.motif_from_position_seq(seq, str(pos_1ix))
+            return {'site_motif': {'motif': motif, 'respos': respos,
+                                   'off_by_one': False}}
+
+
+def _hprd_url(hprd_id, isoform_id, info_type):
```

Comment:
I suspect this is intentional but the isoform_id isn't actually used in the function, it is hardcoded to be 1.